src/lib/y2network/widgets/wireless_expert.rb

```diff
@@ -0,0 +1,137 @@
+require "cwm/common_widgets"
+
+module Y2Network
+  module Widgets
+    # Channel selector widget
+    class WirelessChannel < CWM::ComboBox
+      # @param settings [Y2network::InterfaceConfigBuilder]
+      def initialize(settings)
+        @settings = settings
+
+        textdomain "network"
+      end
+
+      def init
+        self.value = @settings.channel.to_s
+      end
+
+      def store
+        @settings.channel = value.empty? ? nil : value.to_i
+      end
+
+      def label
+        _("&Channel")
+      end
+
+      def opt
+        [:hstretch]
+      end
+
+      def items
+        # FIXME: different protocol has different number of channels, we need to reflect it somehow
+        1.upto(14).map { |c| [c.to_s, c.to_s] }.prepend(["", _("Automatic")])
```

Comment:
Why `1` and `14`? What about using constants with meaningful names?

Reply:
actually there is no better names as it is really number of channels. and calling it "first","second","third" looks a bit strange for me. I will add note there.